### PR TITLE
feat(Treeview): improve a11y

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
* component structure is incorrect
* no accessibility gesture implemented

**What is the chosen solution to this problem?**
Make it accessible following https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-1/treeview-1b.html

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

* props.onClick has been replaced with props.onToggle (more explicit)
* props.structure[].selected has been replaces with props.selectedId

Before
```
import TreeView from '@talend/react-components/lib/TreeView'

function MyComponent(props) {
	...

	const structure = [{ id: 'first', name: 'awesome folder', selected: true }];
	return <TreeView structure={structure} onClick={toggleHandler} ... />
}
```

After
```
import TreeView from '@talend/react-components/lib/TreeView'

function MyComponent(props) {
	...
	const structure = [{ id: 'first', name: 'awesome folder' }];
	return <TreeView structure={structure} selectedId={'first'} onToggle={toggleHandler} ... />
}
```

### CODE REVIEW

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js [hunk 1]
```diff
@@ -112,57 +125,202 @@ class TreeViewItem extends React.Component {
...
+				currentFound = value === this.containerRef;
+				hasNext = !done;
+			}
+		} while (hasNext);
+
+		return nextElement;
+	}
+
+	getPreviousItem() {
+		let previousElement;
```
Comment (jmfrancois) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: the difference between getPreviousItem and getNextItem has lots of repetitions. Could you refactor it please ?

Comment (jmfrancois) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: `width deep structure` example is broken
screen shot 2018-09-05 at 13 55 22

Comment (jsomsanith) [reply packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: `the difference between getPreviousItem and getNextItem has lots of repetitions. Could you refactor it please ?`
--> not really, there are changes that make it not really possible to factor it. The only thing that I can put in common is the selection of all items. I did that

File: packages/components/src/TreeView/TreeView.scss [hunk 2]
```diff
@@ -2,42 +2,26 @@
...
-				fill: $brand-primary;
-				margin-right: 0;
-			}
+		:global(.btn-link .tc-svg-icon) {
+			stroke: $brand-primary;
 		}
 	}
 
-	.tc-treeview-nav {
+	.tc-treeview-ul {
```
Comment (frassinier) [inline packages/components/src/TreeView/TreeView.scss]: Please do not use element in css class name (i.e. `&-list` instead)

**Pull request merged**